inc/document_item.class.php

```diff
@@ -941,6 +941,109 @@ static function getRelationMassiveActionsSpecificities() {
+    *
+    * @since 9.3.1
+    *
+    * @param integer $items_id Object id to restrict on
+    * @param string  $itemtype Type for items to retrieve
+    * @param boolean $noent    Flag to not compute entity informations (see Document_Item::getTypeItemsQueryParams)
+    * @param array   $where    Inital WHERE clause. Defaults to []
+    *
+    * @return array
+    */
+   protected static function getMainTypeItemsQueryParams($items_id, $itemtype, $noent = false, $where = []) {
+      global $DB;
+
+      $item = getItemForItemtype($itemtype);
+      $order_col = $item->getNameField();
+
+      if ($item instanceof CommonDevice) {
+         $order_col = "designation";
+      } else if ($item instanceof Item_Devices) {
+         $order_col = "itemtype";
+      } else if ($itemtype == 'Ticket') {
+         $order_col = 'id';
+      }
+
+      if (!count($where)) {
+         $where = [static::getTable() . '.' . static::$items_id_1  => $items_id];
+      }
+
+      $link_table = $item->getTable();
+      $link_id    = 'items_id';
+
+      $params = [
+         'SELECT'    => [
+            static::getTable() . '.id AS linkid',
+            $link_table . '.*'
+         ],
+         'FROM'      => static::getTable(),
+         'LEFT JOIN' => [
+            $link_table => [
+               'FKEY' => [
+                  static::getTable()   => $link_id,
+                  $link_table          => 'id'
+               ]
+            ]
+         ],
+         'WHERE'     => $where,
+         'ORDER'     => $link_table . '.' . $order_col
+      ];
+
+      $rel_class = static::class;
+      $rel = new $rel_class();
+      if ($rel->maybeDynamic()) {
+         $params['SELECT'][] = static::getTable() . '.is_dynamic';
+      }
+
+      if ($rel->maybeRecursive()) {
+         $params['SELECT'][] = static::getTable() . '.is_recursive';
+      }
+
+      if ($DB->fieldExists(static::getTable(), 'itemtype')) {
+         $params['WHERE'][static::getTable() . '.itemtype'] = $itemtype;
+      }
+
+      if ($item->maybeTemplate()) {
+         $params['WHERE'][$link_table . '.is_template'] = 0;
+      }
+
+      if ($noent === false && $item->isEntityAssign() && $itemtype != Entity::getType()) {
+         $params['SELECT'][] = 'glpi_entities.id AS entity';
+         $params['LEFT JOIN']['glpi_entities'] = [
+            'FKEY'   => [
+               $link_table       => 'entities_id',
+               'glpi_entities'   => 'id'
+            ]
+         ];
+         $params['WHERE'] += getEntitiesRestrictCriteria($link_table, '', '', true);
+         $params['ORDER'] = ['glpi_entities.completename', $params['ORDER']];
+      }
+
+      return $params;
+   }
+
+   /**
+    * Get items for an itemtype
+    *
+    * @since 9.3.1
+    *
+    * @param integer $items_id Object id to restrict on
+    * @param string  $itemtype Type for items to retrieve
+    *
+    * @return DBMysqlIterator
+    */
+   public static function getTypeItems($items_id, $itemtype) {
```

Comment:
Could you explain why it was not possible to override the `getListForItem` and/or 'getListForItemParams' function  instead of redefining about 100 lines of code ?